src/DacpacTool/DacpacTool.csproj

```diff
@@ -7,9 +7,13 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="Microsoft.SqlServer.DACFX" Version="150.4826.1" />
+    <PackageReference Include="Microsoft.SqlServer.DACFX" Version="150.4861.1-preview" />
     <PackageReference Include="System.CommandLine" Version="2.0.0-beta1.20427.1" />
     <PackageReference Include="System.Security.Permissions" Version="4.7.0" />
   </ItemGroup>
 
+  <ItemGroup>
+    <ProjectReference Include="../../sqltoolsservice/src/Microsoft.SqlTools.ManagedBatchParser/Microsoft.SqlTools.ManagedBatchParser.csproj" />
+  </ItemGroup>
+
```

Comment:
Why not just include the files from this folder: https://github.com/microsoft/sqltoolsservice/tree/main/src/Microsoft.SqlTools.ManagedBatchParser/BatchParser

Reply:
Doesn't this need to be a reference to a project, rather than just a list of files?

Reply:
Hmm, yeah, we might be able to get away with `<Compile Include="../../sqltoolsservice/src/Microsoft.SqlTools.ManagedBatchParser/BatchParser/**/*.cs" />`. That would essentially recompile those source files into the `DacpacTool` but I would be fine with that.

Reply:
We might need to add additional `PackageReferences` to make that work though

Reply:
I'll play around with this and see if I can make it work.

Reply:
@ErikEJ @jmezach I've pushed a change that gets really close here -- everything compiles, but 1 test is failing and it's beyond my ability to understand why. It seems to be an issue with the localization resource, but I have the resource included just as `ManagedBatchParser.csproj` does.

Here's the test failure, any thoughts?

```
Test method MSBuild.Sdk.SqlProj.DacpacTool.Tests.PackageBuilderTest.AddPreDeployment_FilesExist threw exception: 
System.Resources.MissingManifestResourceException: Could not find the resource "Microsoft.SqlTools.ManagedBatchParser.Localization.SR.resources" among the resources "Microsoft.SqlTools.ManagedBatchParser.SR.resources" embedded in the assembly "DacpacTool", nor among the resources in any satellite assemblies for the specified culture. Perhaps the resources were embedded with an incorrect name.
  Stack Trace:
      at System.Resources.ManifestBasedResourceGroveler.HandleResourceStreamMissing(String fileName)
   at System.Resources.ManifestBasedResourceGroveler.GrovelForResourceSet(CultureInfo culture, Dictionary`2 localResourceSets, Boolean tryParents, Boolean createIfNotExists)
   at System.Resources.ResourceManager.InternalGetResourceSet(CultureInfo culture, Boolean createIfNotExists, Boolean tryParents)
   at System.Resources.ResourceManager.GetString(String name, CultureInfo culture)
   at Microsoft.SqlTools.ManagedBatchParser.SR.Keys.GetString(String key) in ~/MSBuild.Sdk.SqlProj/sqltoolsservice/src/Microsoft.SqlTools.ManagedBatchParser/Localization/sr.cs:line 409
   at Microsoft.SqlTools.ManagedBatchParser.SR.get_EE_ExecutionError_VariableNotFound() in ~/MSBuild.Sdk.SqlProj/sqltoolsservice/src/Microsoft.SqlTools.ManagedBatchParser/Localization/sr.cs:line 156
   at Microsoft.SqlTools.ServiceLayer.BatchParser.ExecutionEngineCode.BatchParserSqlCmd.GetVariable(PositionStruct pos, String name) in ~/MSBuild.Sdk.SqlProj/sqltoolsservice/src/Microsoft.SqlTools.ManagedBatchParser/BatchParser/ExecutionEngineCode/BatchParserSqlCmd.cs:line 72
   at MSBuild.Sdk.SqlProj.DacpacTool.IncludeVariableResolver.GetVariable(PositionStruct pos, String name) in ~/MSBuild.Sdk.SqlProj/src/DacpacTool/IncludeVariableResolver.cs:line 24
   at Microsoft.SqlTools.ServiceLayer.BatchParser.Parser.ResolveVariables(Token inputToken, Int32 offset, List`1 variableRefs) in ~/MSBuild.Sdk.SqlProj/sqltoolsservice/src/Microsoft.SqlTools.ManagedBatchParser/BatchParser/Parser.cs:line 661
   at Microsoft.SqlTools.ServiceLayer.BatchParser.TextBlock.GetText(Boolean resolveVariables, String& text, LineInfo& lineInfo) in ~/MSBuild.Sdk.SqlProj/sqltoolsservice/src/Microsoft.SqlTools.ManagedBatchParser/BatchParser/TextBlock.cs:line 54
   at MSBuild.Sdk.SqlProj.DacpacTool.ScriptParser.Microsoft.SqlTools.ServiceLayer.BatchParser.ICommandHandler.Go(TextBlock batch, Int32 repeatCount, SqlCmdCommand tokenType) in ~/MSBuild.Sdk.SqlProj/src/DacpacTool/ScriptParser.cs:line 47
   at Microsoft.SqlTools.ServiceLayer.BatchParser.Parser.ExecuteBatch(Int32 repeatCount) in ~/MSBuild.Sdk.SqlProj/sqltoolsservice/src/Microsoft.SqlTools.ManagedBatchParser/BatchParser/Parser.cs:line 204
   at Microsoft.SqlTools.ServiceLayer.BatchParser.Parser.Parse() in ~/MSBuild.Sdk.SqlProj/sqltoolsservice/src/Microsoft.SqlTools.ManagedBatchParser/BatchParser/Parser.cs:line 233
   at MSBuild.Sdk.SqlProj.DacpacTool.ScriptParser.Parse() in ~/MSBuild.Sdk.SqlProj/src/DacpacTool/ScriptParser.cs:line 28
   at MSBuild.Sdk.SqlProj.DacpacTool.ScriptParser.GenerateScript() in ~/MSBuild.Sdk.SqlProj/src/DacpacTool/ScriptParser.cs:line 41
   at MSBuild.Sdk.SqlProj.DacpacTool.PackageBuilder.WritePart(FileInfo file, Package package, String path) in ~/MSBuild.Sdk.SqlProj/src/DacpacTool/PackageBuilder.cs:line 310
   at MSBuild.Sdk.SqlProj.DacpacTool.PackageBuilder.AddScript(FileInfo script, FileInfo outputFile, String path) in ~/MSBuild.Sdk.SqlProj/src/DacpacTool/PackageBuilder.cs:line 299
   at MSBuild.Sdk.SqlProj.DacpacTool.PackageBuilder.AddPostDeploymentScript(FileInfo script, FileInfo outputFile) in ~/MSBuild.Sdk.SqlProj/src/DacpacTool/PackageBuilder.cs:line 89
   at MSBuild.Sdk.SqlProj.DacpacTool.Tests.PackageBuilderTest.AddPreDeployment_FilesExist() in ~/MSBuild.Sdk.SqlProj/test/DacpacTool.Tests/PackageBuilderTests.cs:line 170
```

Reply:
@jeffrosenberg I think that has to do with the <RootNamespace> property set in the project file. I believe that is used when loading the resources. You could try setting it to `Microsoft.SqlTools.ManagedBatchParser` and see if that helps but I'm not sure if I'm happy with that solution to be honest ;).

Reply:
@jmezach sorry, I'm not clear which property you're referring to

Reply:
Ah, sorry, I meant the `RootNamespace` property. That is not set in the project file currently, but you can add it to see if that helps.

Reply:
I think I found another property to fix this -- setting the `LogicalName` for the resource fixed it for me locally, CI tests running now.

Reply:
Looks like that fixes it. Much better solution :).